### ISSUE TITLE
Can I ask One question related to Download Manager

### ISSUE THREAD

**Issue by MasamMahmood** [Yvent/ALDownloadManager#2]
Reference to your download manager, Can I ask one question related to this? if you are available please.

**Comment by Yvent** [comment]
I am sorry.I just saw it. You can ask me now




On 04/17/2020 06:35, MM wrote:

Reference to your download manager, Can I ask one question related to this? if you are available please.

**Comment by MasamMahmood** [comment]
Oh I am sorry, I lately seen it. Actually reference to your download manager I am trying to implement my own custom xib but exactly not getting an idea. 
1) how can I group progress bar - progress? because in your case every file has its own progress as you have added them into array.

I want to achieve multiple pdf files one progress bar.  I am really struggling with that but couldn't achieve.

**Comment by Yvent** [comment]
Rewritten a bit, in the develop branch, you can look at it, answer any questions at any time, hope to help you

**Comment by Yvent** [comment]
Rewritten ALDownloadInfo , in the develop branch, you can look at it, answer any questions at any time, hope to help you

**Comment by MasamMahmood** [comment]
Thank you I have downloaded and thoroughly check, its really useful. But I am stuck in fitting my Xibs because of progress bar thing. Progress is working in console and I am also using `Alamofire`
let me tell you my scenario. 
I have `two json` Appointments and then Appointment Details.  (functions are pullData & DetailData)

- `Appointments` - project name to display 
- `Appointment Details` contains the files for download. 

I am using two `Xibs` 

- One is Main to display only date 
-  Second is Child View depends on appointment count. 

Let me share my project with you. 

[TestDownload.zip](https://github.com/Yvent/ALDownloadManager/files/4592854/TestDownload.zip)

**Comment by Yvent** [comment]
I added a button to your project to control all downloads, you try

[TestDownload.zip](https://github.com/Yvent/ALDownloadManager/files/4596735/TestDownload.zip)

**Comment by MasamMahmood** [comment]
First of all thank you I really appreciate your effort. I know I really bothering you but I am really stuck with thing. 
I have executed your code its working fine but file wise and you displaying file names. 
Actual I want to display 

If I have `two appointments` (I modified json up to 2) and each appointment consist of `16 Files in Appointment Details.`

So it will show 2 appointments and only 2 progress of all files (16 + 16 each). At the moment it displaying separate separate progress bar of every file. (Image attached). 

- **Means one appointment -> 16 files or whatever the count of files -> one progress (of all files).**

- Any possibility we can check the status if its already downloaded or resume.

Thank you once again. 
updated project. (modified appointment.json).

<img width="368" alt="Screen Shot 2020-05-06 at 16 39 02" src="https://user-images.githubusercontent.com/36441313/81377629-998dd400-910e-11ea-9d49-7c40e0e1cfe6.png">

[TestDownload.zip](https://github.com/Yvent/ALDownloadManager/files/4597411/TestDownload.zip)

**Comment by Yvent** [comment]
Two ways
1. Follow the file download to complete the calculation
  8 out of 16 files completed, the total progress is 50%
2. Calculate according to file download progress
  The progress of 16 files is 50%, the total progress is 50%

Which one do you want

**Comment by MasamMahmood** [comment]
First One 
Follow the file download to complete the calculation
  8 out of 16 files completed, the total progress is 50%.

can we mark View circle color change if appointment files not downloaded for any reason any possibility?

**Comment by Yvent** [comment]
try it

[TestDownload.zip](https://github.com/Yvent/ALDownloadManager/files/4598073/TestDownload.zip)

**Comment by MasamMahmood** [comment]
Yes files with progress bar working fine but I want to display appointment which is only 2 appointments and all files progress group into it. 
Not with files names progress. 

Currently your code Image 
<img width="366" alt="Screen Shot 2020-05-08 at 12 25 27" src="https://user-images.githubusercontent.com/36441313/81392392-44aa8780-9127-11ea-8ebc-175b008d524e.png">

Actual I need like this
<img width="368" alt="Screen Shot 2020-05-06 at 16 39 02" src="https://user-images.githubusercontent.com/36441313/81392625-9b17c600-9127-11ea-8601-2e64a001e944.png">

**Comment by MasamMahmood** [comment]
Check updated image. may be this one gives clear picture. 
Pardon me may be I cannot properly communicate it.

Actual result image. 

<img width="368" alt="Screen Shot 2020-05-06 at 16 39 02" src="https://user-images.githubusercontent.com/36441313/81394078-12e6f000-912a-11ea-942d-58812b429704.png">

**Comment by Yvent** [comment]
[TestDownload.zip](https://github.com/Yvent/ALDownloadManager/files/4602427/TestDownload.zip)

**Comment by MasamMahmood** [comment]
Thats same as previous one no change.

**Comment by Yvent** [comment]
You can modify the style you want in the cell

**Comment by MasamMahmood** [comment]
Thank you I know you already did a lot for me. 
I have tried but not successful. 
But is it possible download multiple files with one appointment and one progress ?
Appointment - all files downloads of that appointment - one progress bar only. ( as in Image highlighted in black border)

<img width="368" alt="Screen Shot 2020-05-06 at 16 39 17" src="https://user-images.githubusercontent.com/36441313/81664975-cb71a400-9448-11ea-9111-70278ab8a51e.png">

**Comment by Yvent** [comment]
You said the file name is wrong or the progress is wrong ？

**Comment by MasamMahmood** [comment]
I have set the progress bar accordingly as Last message display. 
But just need to how can I set progress bar smoothly?  Progress bar flickers and starts again with every new file. How it should be smooth and continue ? 
check my updated project. 

[TestDownload.zip](https://github.com/Yvent/ALDownloadManager/files/4624276/TestDownload.zip)

**Comment by Yvent** [comment]
You use multiple progress and one view, all will flash

**Comment by MasamMahmood** [comment]
please can you help me in it ?
